packages/cluster/src/common.js

```diff
@@ -69,8 +76,13 @@ function replacer(string, token, value) {
     return string.replace(regexp, value)
 }
 
-function resolveConfiguration(argv = {}, cache = {}, config = {}) {
-    const resolved = Object.assign({}, defaults, config, cache, argv)
+async function resolveConfiguration(argv = {}) {
+    const file = path.join(clusterDir, argv.name, cacheFile)
+    const cache = JSON.parse(await argv.getCache().read(file))
```

Comment:
`read` returns `null` if the file doesn't exist/cannot be read which works with `JSON.parse`. Could make this more verbose and let the `Cache.read` function throw an error instead.